src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts

```diff
@@ -99,6 +102,37 @@ export class CloneRepoButtonComponent implements OnInit, OnChanges {
         }
     }
 
+    /**
+     * Called when the user clicks on the clone repo button. Sometimes the VCS access token is not present in the user
+     * object, so the resulting repo URL is wrong. This method tries to retrieve the token from the server by requesting
+     * the user object again. This should usually be enough, since this scenario only occurs when starting an exercise
+     * directly after the first time a new user registers via SAML2. In this case Artemis has validated that the
+     * authentication is correct and lets the user log in, but further actions happen in the background that populate
+     * the missing data.
+     *
+     * In case the token could not be retrieved, this method is not called again and the student
+     * is prompted to reload the page. Otherwise, the error would be displayed every time the button is clicked,
+     * including the clicks that make the popover go away (every second click).
+     */
+    onClick() {
+        if (this.versionControlAccessTokenRequired && !this.user.vcsAccessToken && !this.unableToLoadVCSAccessToken) {
+            this.accountService
+                .identity(true)
+                .then((user) => {
+                    this.user = user ?? this.user;
+                    if (!this.user.vcsAccessToken) {
+                        // if still no access token exists after fetching the user object, inform the student that
+                        // something is wrong and that they should try again after reloading the page.
+                        throw new Error();
```

Comment:
I don't think that throwing an empty error is useful here. I would at least add an error message that is understandable to the end user

Reply:
The error is never shown to the user, this `throw` exists only to exit here and let the `catch` block below handle the error case.

The current content of the `catch`-block should be moved into a separate helper method that is called here as well so that the data flow is more clear.